bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Pattern.java

```diff
@@ -187,10 +174,37 @@ public Pattern(Device device, float x1, float y1, float x2, float y2, Color colo
  */
 public Pattern(Device device, float x1, float y1, float x2, float y2, Color color1, int alpha1, Color color2, int alpha2) {
 	super(device);
-	x1 = DPIUtil.autoScaleUp(x1);
-	y1 = DPIUtil.autoScaleUp(y1);
-	x2 = DPIUtil.autoScaleUp(x2);
-	y2 = DPIUtil.autoScaleUp(y2);
+	this.x1 = x1;
+	this.x2 = x2;
+	this.y1 = y1;
+	this.y2 = y2;
+	this.color1 = color1;
+	this.color2 = color2;
+	this.alpha1 = alpha1;
+	this.alpha2 = alpha2;
+	this.isImagePattern = false;
+	initialZoom = DPIUtil.getDeviceZoom();
+	initializeSize(initialZoom);
+}
+
+long getHandle(int zoomLevel) {
+	if (!this.handleMap.containsKey(zoomLevel)) {
+		if (this.isImagePattern) {
+			setImageHandle(image, zoomLevel);
+		} else {
+			initializeSize(zoomLevel);
+		}
+	}
+	return this.handleMap.get(zoomLevel);
+}
+
+private void initializeSize(int zoomLevel) {
+	float x1, y1, x2, y2;
+	long handle;
+	x1 = DPIUtil.autoScaleUp(this.x1, zoomLevel);
+	y1 = DPIUtil.autoScaleUp(this.y1, zoomLevel);
+	x2 = DPIUtil.autoScaleUp(this.x2, zoomLevel);
+	y2 = DPIUtil.autoScaleUp(this.y2, zoomLevel);
```

Comment:
No need to declare the coordinates in a separate line.

```suggestion
	long handle;
	float x1 = DPIUtil.autoScaleUp(this.x1, zoomLevel);
	float y1 = DPIUtil.autoScaleUp(this.y1, zoomLevel);
	float x2 = DPIUtil.autoScaleUp(this.x2, zoomLevel);
	float y2 = DPIUtil.autoScaleUp(this.y2, zoomLevel);
```

Also, if you could rename the fields that are called exactly the same, you wouldn't obscure them in this method. Obscuring fields is a dangerous practice, please avoid it when possible.